herald/herald/Sensor/BLE/BLEReceiver.swift

```diff
@@ -933,6 +936,10 @@ class ConcreteBLEReceiver: NSObject, BLEReceiver, BLEDatabaseDelegate, CBCentral
         // Wrote characteristic -> Scan again
         let device = database.device(peripheral, delegate: self)
         logger.debug("didWriteValueFor (device=\(device),error=\(String(describing: error)))")
+        // OpenTrace -> read -> write -> disconnect
+        if device.legacyPayloadCharacteristic != nil, device.signalCharacteristic == nil {
```

Comment:
Possible for a device to support a legacy payload but also have herald and a signal characteristic.

Reply:
This will only happen if target is running a legacy app, and a seperate Herald app. In which case, Herald will not capture the payload from legacy app. It is an unlikely scenario which adds complexity to Herald because a Herald app with OpenTrace support will appear the same way (Herald signal, Herald payload, Legacy payload characteristics). This is a transitional feature, let's revisit if the need arises, but rather keep it simple for now. Given the intricate nature of protocols, Legacy in this instance really means just OpenTrace.